### PR TITLE
Added some documentation

### PR DESCRIPTION
**Pull request by dephillipsmichael**
Added some documentation

### CODE REVIEW

**Review by syoung-smallwisdom — APPROVED** [review]
👍

File: Research/Research/RSDResult.swift [hunk 1]
```diff
@@ -77,6 +77,8 @@ public protocol RSDTaskResult : RSDResult, RSDAnswerResultFinder {
     var stepHistory: [RSDResult] { get set }
     
     /// A list of all the asynchronous results for this task. The list should include uniquely identified results.
+    /// The step history is used to describe the path you took to get to where you are going whereas,
```
Comment (Erin-Mounts) [inline Research/Research/RSDResult.swift]: comma should come after 'going' not after 'whereas'

File: Research/Research/RSDResult.swift [hunk 2]
```diff
@@ -77,6 +77,8 @@ public protocol RSDTaskResult : RSDResult, RSDAnswerResultFinder {
...
+    /// the asynchronous results include any canonical result that are independent of path.
```
Comment (Erin-Mounts) [inline Research/Research/RSDResult.swift]: canonical result**s**

File: Research/Research/RSDResult.swift [hunk 3]
```diff
@@ -137,6 +139,8 @@ extension RSDTaskResult  {
     }
     
     /// Append the async results with the given result, replacing the previous instance with the same identifier.
+    /// The step history is used to describe the path you took to get to where you are going whereas,
```
Comment (Erin-Mounts) [inline Research/Research/RSDResult.swift]: going whereas, --> going, whereas

File: Research/Research/RSDResult.swift [hunk 4]
```diff
@@ -137,6 +139,8 @@ extension RSDTaskResult  {
...
+    /// the asynchronous results include any canonical result that are independent of path.
```
Comment (Erin-Mounts) [inline Research/Research/RSDResult.swift]: canonical result**s**

File: Research/Research/RSDTaskPath.swift [hunk 5]
```diff
@@ -285,6 +285,8 @@ public final class RSDTaskPath : NSObject, NSCopying {
     }
     
     /// Append the async results with the given result, replacing the previous instance with the same identifier.
+    /// The step history is used to describe the path you took to get to where you are going whereas,
```
Comment (Erin-Mounts) [inline Research/Research/RSDTaskPath.swift]: ditto

File: Research/Research/RSDTaskPath.swift [hunk 6]
```diff
@@ -285,6 +285,8 @@ public final class RSDTaskPath : NSObject, NSCopying {
...
+    /// the asynchronous results include any canonical result that are independent of path.
```
Comment (Erin-Mounts) [inline Research/Research/RSDTaskPath.swift]: and ditto

**Comment by dephillipsmichael** [comment]
Updated for grammar

**Review by Erin-Mounts — APPROVED** [review]

**Pull request merged**